File: pension-management/src/main/java/com/pension/pension_management/entity/Contribution.java
package com.pension.pension_management.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "contributions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Contribution {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    @Column(name = "member_number", nullable = false, length = 20)
    private String memberNumber;

    @Column(nullable = false, precision = 15, scale = 2)
    private BigDecimal amount;

    // FIXED: This is the main issue
    @Column(name = "contribution_date", nullable = false, updatable = false)
    private LocalDateTime contributionDate = LocalDateTime.now();

    @Column(length = 100)
    private String paymentMethod;

    @Column(length = 255)
    private String referenceNumber;

    @Column(length = 500)
    private String notes;
}

File: pension_app/lib/models/contribution.dart
class Contribution {
  final int id;
  final String memberNumber;
  final String fullName;
  final double amount;
  final String? contributionDate;
  final String? paymentMethod;
  final String? referenceNumber;
  final String? notes;

  Contribution({
    required this.id,
    required this.memberNumber,
    required this.fullName,
    required this.amount,
    this.contributionDate,
    this.paymentMethod,
    this.referenceNumber,
    this.notes,
  });

  factory Contribution.fromJson(Map<String, dynamic> json) {
    return Contribution(
      id: json['id'],
      memberNumber: json['memberNumber'],
      fullName: json['fullName'] ?? '',
      amount: double.parse((json['amount'] ?? 0).toString()),
      contributionDate: json['contributionDate'],
      paymentMethod: json['paymentMethod'],
      referenceNumber: json['referenceNumber'],
      notes: json['notes'],
    );
  }
}